### ISSUE TITLE
some questions about the figure came from the On Deep Learning-Based Channel Decoding.ipynb

### ISSUE THREAD

**Issue by EllieElliegogo** [gruberto/DL-ChannelDecoding#1]
After reading your article On Deep Learning-Based Channel Decoding,I noticed that in the figure.3 all BER lines of NN are above  that of MAP.while I saw the BER lines of the NN and the MAP in the [https://github.com/gruberto/DL-ChannelDecoding/blob/master/notebooks/On%20Deep%20Learning-Based%20Channel%20Decoding.ipynb](url),I found that the NN line is sometimes below the MAP one.
I wonder why this could happen? would you please tell me some information? I am learning the classic Polar Codes for just half year and totally a fresh man on deep learning,so I'd be very appreciate if you could help me .
thanks!

**Comment by gruberto** [comment]
That should be a simulation artifact. Simulating with more codewords and more SNR points will help to fix this issue.

**Comment by qindadi** [comment]
hello.
    When I ran the code, I noticed a slight difference with your results. There are training procedures and testing procedures, but the number of error codes (nb_errors) is always zero and the graph is missing nn line. I'd be very appreciate if you could help me .
![figure_1-1](https://user-images.githubusercontent.com/35325855/39811385-0c198124-53bb-11e8-834c-536e698f1967.png)

**Comment by gruberto** [comment]
Hi. In the test section, the performance of a MAP decoder and a NN decoder is evaluated. In order to obtain bit error rates even for high Eb/N0, you need to simulate enough code words (num_words). If you simulate with too less words, no error will occur. This does not affect the MAP curve because this curve is loaded from a file. I hope, I could solve your problem. If not, please give more details about your problem.

**Comment by qindadi** [comment]
First of all, thank you for your answer for me. I try to change the code words (num_words) from 10000 to 80000, but still zero error number (nb_errors), I don't understand (1) : this code words (num_words) is how to produce. (2) the thesis training set and testing set of data is only 256, why still need to 2 ^ 16 training wheels (epochs). (3) : What does nb_bits stand for? Why is it the square value of code words(num_words)? (4): why I have only three columns in figure 1, instead of four columns like yours (not connected to this column)
![2018-05-12 13-44-52](https://user-images.githubusercontent.com/35325855/39954431-f91699f2-55f1-11e8-86c7-8e4b7d601afc.png)

**Comment by qindadi** [comment]
![2018-05-12 14-11-42](https://user-images.githubusercontent.com/35325855/39954442-27f60668-55f2-11e8-995c-d0a3800972ff.png)

**Comment by SebastianCa** [comment]
Unfortunately, without further look into your modifications ragarding the source code it is impossible to trace the problem. The decoder seems to make no errors (which is impossible), thus, either a problem with the additive noise or the codeword generation (all-zero?) may occur.

Regarding your questions:
1.) num_words is simply the amount of Monte Carlo trials for the BER calculation (size of the validation set)
2.) The same 256 codewords (i.e., classes)  are used 2^16 times for training, but always with different noise realization (ensured by AWGN-layer)
3.) nb_bits is the amount of transmitted bits during the Monte Carlo BER simulation and thus the BER=nb_errors/nb_bits; I cannot see where it is the square value of num_words
4.) I do not understand what you refer to.
